sound/soc/sof/intel/hda-dai.c

```diff
@@ -225,6 +195,8 @@ static int hda_link_dma_hw_params(struct snd_pcm_substream *substream,
 		snd_soc_dai_set_dma_data(cpu_dai, substream, (void *)hext_stream);
 	}
 
+	hstream = &hext_stream->hstream;
+	stream_tag = hstream->stream_tag;
```

Comment:
You don't really need the `hstream` here:
```c
stream_tag = hext_stream->hstream.stream_tag;
```
or use it at line 228:
```c
if (hstream->direction == SNDRV_PCM_STREAM_PLAYBACK) {
```

Reply:
can we let this one go? all this code will eventually go away with the ops anyway